proxy/mempool/mempool_schedule.py

```diff
@@ -285,11 +296,42 @@ def _add_tx_to_sender_pool(self, sender_pool: MPSenderTxPool, tx: MPTxRequest) -
         if sender_pool.len_tx_nonce_queue == 1:
             self._sender_pool_dict[sender_pool.sender_address] = sender_pool
 
+        self._sender_pool_heartbeat_queue.try_add(sender_pool)
+
     def _drop_tx_from_sender_pool(self, sender_pool: MPSenderTxPool, tx: MPTxRequest) -> None:
         LOG.debug(f'Drop tx {tx.sig} from pool {sender_pool.sender_address}')
         sender_pool.drop_tx(tx)
         self._tx_dict.pop_tx(tx)
 
+    def drop_expired_sender_pools(self, eviction_timeout_sec: int) -> None:
+        threshold = int(time()) - eviction_timeout_sec
+        LOG.debug(f'Try to drop sender pools with heartbeat below {threshold}')
+
+        while True:
+            sender_pool =  self.peek_lowest_sender_pool_heartbeat()
```

Comment:
`sender_pool = self._sender_pool_heartbeat_queue[self._top_index]`
If your queue is reversed